community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/commands/expressions/StringFunctions.scala

```diff
@@ -177,14 +188,14 @@ case class SplitFunction(orig: Expression, separator: Expression)
     val separatorVal = asString(separator(m, state))
 
     if (origVal == null || separatorVal == null) {
-      Values.NO_VALUE
+      NO_VALUE
     } else {
       if (separatorVal.length > 0) {
-        VirtualValues.fromArray(Values.stringArray(split(Vector.empty, origVal, 0, separatorVal).toArray:_*))
```

Comment:
I agree. I stopped at methods that requires regex, since that is trickier to implement for `UTF8StringValue`. We could of course just have a default implementation that uses strings, but leave that for later.